components/textfield/stories/template.js

```diff
@@ -50,8 +51,13 @@ export const Template = ({
 		console.warn(e);
 	}
 
-	if (isInvalid) iconName = "Alert";
-	else if (isValid) iconName = "Checkmark";
+	if (isInvalid) {
+		iconName = "Alert";
+		iconSet = "workflow";
+	} else if (isValid) {
+		iconName = "Checkmark";
+		iconSet = "ui";
```

Comment:
Why pass in iconName and iconSet as variables? It seems like they're being hardcoded here? I'm also curious why this has an isInvalid and isValid flag? Should we just use one or the other and use negation to toggle between them?

Reply:
It looks like the icon is an arg because the Search component passes in a third option, `iconName: "Magnify"`. If we wanted to refactor to remove those args that allow choosing the icon for this story, I think we would need to add a different arg like`isSearch`?

isInvalid and isValid are for toggling those classes for the alert or the checkmark, and the third state is not set, which doesn't display either. It looks like Datepicker is also doing it this way. We could use a select list of options with undefined, is-valid, or is-invalid, but I think it's probably good to be able to test when both classes are present to make sure that the error always overrides valid.

Reply:
If search passes in Magnify and it's listed as isInvalid - does the icon get overridden?

Reply:
I suppose search can't be invalid - just trying to think what states things could get in to